plugin/pkg/scheduler/algorithm/priorities/interpod_affinity.go

```diff
@@ -102,10 +103,24 @@ func (p *podAffinityPriorityMap) processTerm(term *api.PodAffinityTerm, podDefin
 	}
 }
 
-func (p *podAffinityPriorityMap) processTerms(terms []api.WeightedPodAffinityTerm, podDefiningAffinityTerm, podToCheck *api.Pod, fixedNode *api.Node, multiplier int) {
+func (p *podAffinityPriorityMap) processTerms(terms[] api.WeightedPodAffinityTerm, podDefiningAffinityTerm, podToCheck *api.Pod, fixedNode *api.Node, multiplier int, podlister algorithm.PodLister) {
 	for i := range terms {
 		term := &terms[i]
-		p.processTerm(&term.PodAffinityTerm, podDefiningAffinityTerm, podToCheck, fixedNode, float64(term.Weight*multiplier))
+
+
+		// A crude implementation of on-node pod label selector scoring.
+		selector,_ := unversioned.LabelSelectorAsSelector(term.PodAffinityTerm.LabelSelector)
```

Comment:
@wojtek-t As a first hack/pass  I tried scanning all the pods which match the selector - and increasing node scores if one matching pod is found. This  Seems to mostly effect the behaviour we want.   It adds more subtelety to the scoring algorithm though ... not sure the right way to weight .